sshd-common/src/main/java/org/apache/sshd/common/future/WaitableFuture.java

```diff
@@ -65,6 +66,19 @@ default boolean await(long timeout, TimeUnit unit) throws IOException {
         return await(unit.toMillis(timeout));
     }
 
+    /**
+     * Wait for the asynchronous operation to complete with the specified timeout.
+     *
+     * @param timeout   The maximum duration to wait
+     * @return {@code true} if the operation is completed.
+     * @throws IOException if failed - specifically {@link java.io.InterruptedIOException}
+     *                     if waiting was interrupted
+     * @see #await(long)
+     */
+    default boolean await(Duration timeout) throws IOException {
+        return await(timeout.toMillis());
```

Comment:
In other locations the  code interprets `null` as *forever* - let's maintain this convention (or fix the other code - `ClientChannel#waitFor`) - see similar previous comment